Modules/Core/Common/include/itkMacro.h

```diff
@@ -394,6 +394,15 @@ namespace itk
     static_assert(false, "Replace deprecated ITK_DISALLOW_COPY_AND_ASSIGN with modern ITK_DISALLOW_COPY_AND_MOVE")
 #endif
 
+#if __cplusplus < 202002L
+#  define ITK_DEFINE_INEQUAL_OPERATOR_MEMBER_FUNCTION(TypeName)                                                        \
+    bool operator!=(const Self & other) const { return !(this->operator==(other)); }                                   \
```

Comment:
With
```c++
bool operator!=(const Self & rhs) const { return !(*this == rhs); }
```
the macro might also work for `friend` equality operators.

Reply:
Good idea @Leengit However, I want to just make one step at a time. This may be the next step.

My initial aim is just to make C++20 happy (PR #2589 by @darrent1974), while preserving the original (ITK 5.2.0) behavior as much as possible.

Ideally I would like to move entirely from "member function" equality operators to `friend` equality operator, but I don't know if it is feasible. It would help if an automatic tool like clang-tidy could do the job!